src/libutil/logging.hh

```diff
@@ -4,6 +4,8 @@
 #include "error.hh"
 #include "config.hh"
 
+#include <nlohmann/json.hpp>
```

Comment:
We should never include `nlohmann/json.hpp` in any header file, since (IIRC) just including it adds about half a second of compilation time per compilation unit. If really needed, you can use `json_fwd.hpp`.